### PR TITLE
Fixing docker_config script to append multiple insecure registries

### PR DESCRIPTION
**Pull request by LalatenduMohanty**
Fixes issue #29

Signed-off-by: Lalatendu Mohanty lmohanty@redhat.com

### CODE REVIEW

**Comment by LalatenduMohanty** [comment]
Donot merge till https://github.com/projectatomic/adb-utils/pull/46 is merged as it is reusing script get_ip_address

**Comment by LalatenduMohanty** [comment]
@hferentschik  comments?

**Comment by praveenkumar** [comment]
@LalatenduMohanty this script is part of openshift service and not going to used as independent. If I understand correctly about @hferentschik  issue #29 (as per 2nd comment) he tried to put some insecure registry to docker config and our old script inverted it. Current script have well implemented if we are going to use it independently like 

```
docker-config <registry>
```

but our current use case is only to enable internal docker registry for OpenShift and expose VM registry to host. We only require some part of script in that case.

@hferentschik correct me if I am wrong.

**Comment by hferentschik** [comment]
> but our current use case is only to enable internal docker registry for OpenShift and expose VM registry to host. We only require some part of script in that case.

There is also use case of adding a user specific registry, for example in company internal registry or similar. If the script is nicely written, one could easily call this script as part of the provisioning (instead of using sed, etc). That said, afaiu, we have been discussing that the service-manager plugin will expose an option to add registries via a config option. For the user that is even better. Implementation wise I could imagine that the plugin would like to call `docker-config`. 

Bottom line, there needs to way an easy way for the user to add custom registries. Being able to call some script in a shell provisioner is imo the minimum. An option in service-manager is even better.

@bexelbie I am correct that we discussed this option, right?

**Comment by hferentschik** [comment]
@praveenkumar 

> this script is part of openshift service and not going to used as independent

Sure, but what is wrong with making it more generically usable with little extra effort?

File: services/openshift/scripts/docker_config [hunk 1]
```diff
@@ -1,28 +1,25 @@
 #!/usr/bin/env bash
 
-# Stop OpenShift Releated Containers
+#. /opt/adb/openshift/get_ip_address
```
Comment (praveenkumar) [inline services/openshift/scripts/docker_config]: There is a typo `#` should not be present because it making this source include as comment.

File: services/openshift/scripts/docker_config [hunk 2]
```diff
@@ -32,8 +29,43 @@ while [ $# -gt 0 ]; do
...
-sed -i.back "/# INSECURE_REGISTRY=*/c\INSECURE_REGISTRY=\"\"" /etc/sysconfig/docker
-if ! grep -qE 172.30.0.0.*${ip} /etc/sysconfig/docker; then
-    sed -i.back "s/INSECURE_REGISTRY=\"\(.*\)\"/INSECURE_REGISTRY=\"\1 --insecure-registry 172.30.0.0\/16 --insecure-registry $ip\"/" /etc/sysconfig/docker
+function init_insecure_reg()
+{
+    if grep -q '# INSECURE_REGISTRY' ${docker_conf_file}
+    then
+        sed -i.orig "/# INSECURE_REGISTRY=*/c\INSECURE_REGISTRY=\"\"" ${docker_conf_file}
+        # Get machine IP address
+        local local_ip=get_ip_address
```
Comment (praveenkumar) [inline services/openshift/scripts/docker_config]: This will again need changed to

```
local local_ip=$(get_ip_address)
```

File: services/openshift/scripts/docker_config (hunk 1 above)
Comment (LalatenduMohanty) [inline services/openshift/scripts/docker_config]: @praveenkumar done

**Comment by praveenkumar** [comment]
LGTM

File: services/openshift/scripts/docker_config [hunk 3]
```diff
@@ -32,8 +29,43 @@ while [ $# -gt 0 ]; do
...
+    fi
+}
+
+#Main
+init_insecure_reg
+if ! test -z $ip ; then
+    append_insecure_reg ${ip}
+fi
+
+#Restart the docker demon if $is_restart_req is set to true
```
Comment (hferentschik) [inline services/openshift/scripts/docker_config:68]: nice with some comments :-)
